sdktests/server_side_persistence_base.go

```diff
@@ -119,58 +134,560 @@ func newServerSidePersistentTests(
+			Type:   s.persistentStore.Type(),
+			DSN:    s.persistentStore.DSN(),
+			Prefix: o.Some(customPrefix),
+		})
+		persistence.SetCache(servicedef.SDKConfigPersistentCache{
+			Mode: servicedef.CacheModeOff,
+		})
+
+		client := NewSDKClient(t, persistence)
+
+		require.Never(
+			t,
+			checkForUpdatedValue(t, client, "flag-key", ldcontext.New("user-key"),
+				ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default")),
+			time.Millisecond*100,
+			time.Millisecond*20,
+			"flag value was updated, but it should not have been",
+		)
+
+		require.NoError(t, s.persistentStore.WriteMap(customPrefix, "features", s.initialFlags))
+
+		pollUntilFlagValueUpdated(t, client, "flag-key", ldcontext.New("user-key"),
+			ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
 	})
-	persistence.SetCache(servicedef.SDKConfigPersistentCache{
-		Mode: servicedef.CacheModeOff,
+
+	t.Run("daemon mode", func(t *ldtest.T) {
+		persistence := NewPersistence()
+		persistence.SetStore(servicedef.SDKConfigPersistentStore{
+			Type: s.persistentStore.Type(),
+			DSN:  s.persistentStore.DSN(),
+		})
+		persistence.SetCache(servicedef.SDKConfigPersistentCache{
+			Mode: servicedef.CacheModeOff,
+		})
+		context := ldcontext.New("user-key")
+
+		s.runWithEmptyStore(t, "ignores database initialization flag", func(t *ldtest.T) {
+			client := NewSDKClient(t, persistence)
+
+			h.RequireEventually(t, func() bool {
+				result := client.EvaluateFlag(t, servicedef.EvaluateFlagParams{
+					FlagKey:      "flag-key",
+					Context:      o.Some(context),
+					ValueType:    servicedef.ValueTypeAny,
+					DefaultValue: ldvalue.String("default"),
+					Detail:       true,
+				})
+
+				return result.Reason.IsDefined() &&
+					result.Reason.Value().GetErrorKind() == ldreason.EvalErrorFlagNotFound
+			}, time.Second, time.Millisecond*20, "flag was found before it should have been")
+
+			require.NoError(t, s.persistentStore.WriteMap(s.defaultPrefix, "features", s.initialFlags))
+			pollUntilFlagValueUpdated(t, client, "flag-key", context,
+				ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
+		})
+
+		s.runWithEmptyStore(t, "can disable cache", func(t *ldtest.T) {
+			require.NoError(t, s.persistentStore.WriteMap(s.defaultPrefix, "features", s.initialFlags))
+
+			client := NewSDKClient(t, persistence)
+			pollUntilFlagValueUpdated(t, client, "flag-key", context,
+				ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
+
+			// Completely reset the database so there are no valid flag definitions
+			require.NoError(t, s.persistentStore.Reset())
+
+			h.RequireEventually(t,
+				checkForUpdatedValue(t, client, "flag-key", context,
+					ldvalue.String("fallthrough"), ldvalue.String("default"), ldvalue.String("default")),
+				time.Second, time.Millisecond*20, "flag value was NOT updated after cache TTL")
```

Comment:
I'm getting mixed messages between the assertion message and the test description.